include/my_global.h

```diff
@@ -1234,4 +1234,22 @@ static inline double rint(double x)
 #undef __GNUG__
 #endif
 
+/*
+  Provide defaults for the CPU cache line size, if it has not been detected by
+  CMake using getconf
+*/
+#if !defined(CPU_LEVEL1_DCACHE_LINESIZE) || CPU_LEVEL1_DCACHE_LINESIZE == 0
+  #if CPU_LEVEL1_DCACHE_LINESIZE == 0
+    #undef CPU_LEVEL1_DCACHE_LINESIZE
+  #endif
```

Comment:
This looks a bit irrational. Why not:
#if CPU_LEVEL1_DCACHE_LINESIZE == 0
#undef CPU_LEVEL1_DCACHE_LINESIZE
#endif
#if !defined(CPU_LEVEL1_DCACHE_LINESIZE)
...
#endif